### PR TITLE
Update group menu styles

### PR DESCRIPTION
**Pull request by robertknight**
This updates the styling of menu items following a design review. See commit messages for details. The changes mostly pertain only to features of menus seen in the groups menu. The one remaining thing that has not been done here is to update the submenu item icons and "New private group" icons to match the mocks.

Mocks: https://sketch.cloud/s/D0OMk/a/EbnxQA/play

**Before (submenu open and item hovered):**

Screenshot 2019-06-18 13 15 38

**After (submenu open):**

Screenshot 2019-06-18 13 04 21

**After (submenu open, item hovered):**

Screenshot 2019-06-18 13 04 32

### CODE REVIEW

**Comment by lyzadanger** [comment]
@robertknight 

> The one remaining thing that has not been done here is to update the submenu item icons and "New private group" icons to match the mocks.

I don't see this icon in the mock you linked to. Do you happen to have a link to a mock that shows this icon?

**Comment by lyzadanger** [comment]
I'm having a little trouble with discerning the hovers; with a few submenus open it makes it hard to tell what I'm currently hovering on:

subtle-hovers

The hover state is so subtle as to be not really a hover state to me...

**Comment by lyzadanger** [comment]
@robertknight I DO like the new external-link icon!

**Comment by lyzadanger** [comment]
Here's what the hover looks like in red:

less-subtle-hover

**Comment by lyzadanger** [comment]
Ooh! I like the dark-gray hover! The only thing bugging me _now_ is that the group names don't have a hover:

better-hover

**Comment by lyzadanger** [comment]
Sorry for the go-around on this, but what I was thinking with the `blank` idea was that any provided `icon` prop would be truthy. The options would be:

* string name of registered `SVGIcon`
* URL to image
* literal string `blank`

That would obviate the need—I think?—to do any `null`-specific testing in the component.

If no value is provided for the prop, take the falsey path and don't render an icon at all (real or blank).

What do you think?

File: src/styles/sidebar/components/group-list-item.scss [hunk 1]
```diff
@@ -8,4 +12,8 @@
   // Align the left edge of the footer text with menu item labels above.
   padding-left: 35px;
   white-space: normal;
+
+  // Override default hyphenation which is applied to all `p` elements in the
+  // app.
+  hyphens: none;
```
[the remark this answers is not in the document]
Comment (robertknight) [reply src/styles/sidebar/components/group-list-item.scss]: The word "specific" got hyphenated and split over two lines in the message "This group is restricted to specific URLs.", which I think looked a bit ugly and unnecessary.

**Pull request merged**